nextest-runner/src/reporter/displayer.rs

```diff
@@ -29,12 +29,172 @@ use serde::Deserialize;
 use std::{
     borrow::Cow,
     cmp::Reverse,
-    fmt::{self, Write as _},
+    fmt::{self, Formatter, Write as _},
     io,
     io::{BufWriter, Write},
+    str::FromStr,
     time::Duration,
 };
 
+/// When to display test output in the reporter.
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq)]
+#[cfg_attr(test, derive(test_strategy::Arbitrary))]
+pub struct TestOutputDisplayStreams {
+    /// When to display the stdout output of a test in the reporter
+    pub stdout: Option<TestOutputDisplay>,
+    /// When to display the stderr output of a test in the reporter
+    pub stderr: Option<TestOutputDisplay>,
+}
+
+impl TestOutputDisplayStreams {
+    /// Create a `TestOutputDisplayStreams` with both stdout and stderr set to `Immediate`
+    pub fn create_immediate() -> Self {
+        Self {
+            stdout: Some(TestOutputDisplay::Immediate),
+            stderr: Some(TestOutputDisplay::Immediate),
+        }
+    }
+    /// Create a `TestOutputDisplayStreams` with both stdout and stderr set to `Final`
+    pub fn create_final() -> Self {
+        Self {
+            stdout: Some(TestOutputDisplay::Final),
+            stderr: Some(TestOutputDisplay::Final),
+        }
+    }
+    /// Create a `TestOutputDisplayStreams` with both stdout and stderr set to `ImmediateFinal`
+    pub fn create_immediate_final() -> Self {
+        Self {
+            stdout: Some(TestOutputDisplay::ImmediateFinal),
+            stderr: Some(TestOutputDisplay::ImmediateFinal),
+        }
+    }
+    /// Create a `TestOutputDisplayStreams` with both stdout and stderr set to `Never`
+    pub fn create_never() -> Self {
+        Self {
+            stdout: Some(TestOutputDisplay::Never),
+            stderr: Some(TestOutputDisplay::Never),
+        }
+    }
+
+    /// Which output streams should be output immediately
+    ///
+    /// # Returns
```

Comment:
We generally don't use a `# Returns` header here.

Also I'd rather this return a `DisplayOutput` rather than `Option<DisplayOutput>`, to avoid representing the invalid state where `DisplayOutput` is set to `false` for both.